ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py

```diff
@@ -0,0 +1,11 @@
+import ivy
+
+def numpy_diagonal(a, offset, axis1, axis2):
```

Comment:
The name of the function should be just `diagonal` (~~numpy_~~ diagonal). That's why you get the key error.